src/game/CObjBase.cpp

```diff
@@ -1971,6 +2017,59 @@ bool CObjBase::r_Verb( CScript & s, CTextConsole * pSrc ) // Execute command fro
                 m_TooltipData.emplace_back(new CClientTooltip(clilocid, sLocArgs.GetPtr()));
             }
             break;
+        case OV_REMOVECLILOC:
+        	// Remove cliloc in @ClientTooltip trigger
+        	{
+                tchar * ppLocArgs[256];
+                Str_ParseCmds(s.GetArgRaw(), ppLocArgs, CountOf(ppLocArgs), ",");
+        		dword clilocid = Exp_GetVal(ppLocArgs[0]);
+        		for (size_t i = 0; i < m_TooltipData.size(); i++)
+        		{
+        			CClientTooltip* ct = m_TooltipData[i].get();
+        			if (ct->m_clilocid == clilocid)
+        			{
+						if ( g_Cfg.m_iDebugFlags & DEBUGF_SCRIPTS )
+							g_Log.EventDebug("SCRIPT: removecliloc(%u)\n", clilocid);
+        				m_TooltipData.erase(m_TooltipData.begin() + i);
+        				//I did a break, but if same tooltip added as script and default together there could be more than one for same cliloc so we need to push to check if there is another.
+        				//break;
+					}
+				}
+			}
+			break;
+		case OV_REPLACECLILOC:
+			// Replace cliloc in @ClientTooltip trigger
+			{
+				tchar * ppLocArgs[256];
+                int qty = Str_ParseCmds(s.GetArgRaw(), ppLocArgs, CountOf(ppLocArgs), ",");
+                dword clilocid = Exp_GetVal(ppLocArgs[0]);
+                
+                CSString sLocArgs;
+                for (int y = 1 ; y < qty; ++y )
+                {
+                    if ( sLocArgs.GetLength() )
+                        sLocArgs += "\t";
+
+                    if ((*ppLocArgs[y] == '\0') || !strncmp(ppLocArgs[y], "NULL", 4))
+                        sLocArgs += " ";
+                    else
+                        sLocArgs += ppLocArgs[y];
+                }
+                
+                for (size_t i = 0; i < m_TooltipData.size(); i++)
+                {
+                	CClientTooltip* ct = m_TooltipData[i].get();
+                	if (ct->m_clilocid == clilocid)
+                	{
+		                if ( g_Cfg.m_iDebugFlags & DEBUGF_SCRIPTS )
+        	    	        g_Log.EventDebug("SCRIPT: replacecliloc(%u,'%s')\n", clilocid, sLocArgs.GetPtr());
+        	    	    m_TooltipData.erase(m_TooltipData.begin() + i);
+        	    	    m_TooltipData.emplace(m_TooltipData.begin() + i, new CClientTooltip(clilocid, sLocArgs.GetPtr()));
```

Comment:
Another tip: vector's emplace method is very nice also because you can avoid manually creating the object, you can directly pass the constructor's arguments like this:
`m_TooltipData.emplace(m_TooltipData.begin() + i, clilocid, sLocArgs.GetPtr());`
This results in shorter code and lets the compiler use the best way to create the object (which isn't by manually using `new` since in this case we are using an unique_ptr, it should be with std::make_unique)

Reply:
I haven't known about it, I am updating the code quickly.

Reply:
When I changed emplace with that way vs gives that error;
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2664	'std::unique_ptr<CClientTooltip,std::default_delete<CClientTooltip>>::unique_ptr(const std::unique_ptr<CClientTooltip,std::default_delete<CClientTooltip>> &)': cannot convert argument 1 from 'dword' to 'CClientTooltip *' (compiling source file C:\Users\user\Documents\GitHub\Source-X\src\game\CObjBase.cpp)	spheresvr	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.25.28610\include\xmemory	671	
```

Reply:
You're right, std::unique_ptr constructor doesn't accept an argument list, so you should directly pass a `std::make_unique<CClientTooltip>(clilocid, sLocArgs.GetPtr())`

Reply:
I replaced the code with std::make_unique